fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3237,6 +3273,46 @@ public void clearHeaderUserField(@Nonnull String userField) {
         context.asyncToSync(FDBStoreTimer.Waits.WAIT_EDIT_HEADER_USER_FIELD, clearHeaderUserFieldAsync(userField));
     }
 
+    /**
+     * Update the RecordCount to have a new state.
+     * <p>
+     *     The state can go from {@code READABLE} to {@code WRITE_ONLY} and vice-versa, and it can always be set to
+     *     {@code DISABLED}, but it cannot transition out of {@code DISABLED}.
```

Comment:
I assume this is a consequence of not adding a build record count key or some equivalent? That seems fine given that the key is effectively deprecated anyway. I'm not _totally_ sure I see the vision for `WRITE_ONLY` here

Reply:
ok, I will elaborate. I thought I had explained, that but clearly I didn't do so here, and repeating here would be good.